marconi/src/Marconi/Index/ScriptTx.hs

```diff
@@ -275,21 +300,35 @@ instance Queryable ScriptTxHandle where
 
 instance Rewindable ScriptTxHandle where
   rewindStorage
-    :: SlotNo
+    :: ChainPoint
     -> ScriptTxHandle
     -> IO (Maybe ScriptTxHandle)
-  rewindStorage sn h@(ScriptTxHandle c _) = do
+  rewindStorage (ChainPoint sn   _) h@(ScriptTxHandle c _) = do
      SQL.execute c "DELETE FROM script_transactions WHERE slotNo > ?" (SQL.Only sn)
      pure $ Just h
+  rewindStorage ChainPointAtGenesis h@(ScriptTxHandle c _) = do
+     SQL.execute_ c "DELETE FROM script_transactions"
+     pure $ Just h
+
+
+-- For resuming we need to provide a list of points where we can resume from.
+
+instance Resumable ScriptTxHandle where
+  resumeFromStorage h = do
+    es <- Storable.getStoredEvents h
+    -- The ordering here matters. The node will try to find the first point in the
+    -- ledger, then move to the next and so on, so we will send the latest point
+    -- first.
+    pure $ map chainPoint es ++ [ChainPointAtGenesis]
 
 open :: FilePath -> Depth -> IO ScriptTxIndexer
 open dbPath (Depth k) = do
   c <- SQL.open dbPath
-  SQL.execute_ c "CREATE TABLE IF NOT EXISTS script_transactions (scriptAddress TEXT NOT NULL, txCbor BLOB NOT NULL, slotNo INT NOT NULL)"
+  SQL.execute_ c "CREATE TABLE IF NOT EXISTS script_transactions (scriptAddress TEXT NOT NULL, txCbor BLOB NOT NULL, slotNo INT NOT NULL, blockHash BLOB NOT NULL)"
   SQL.execute_ c "CREATE INDEX IF NOT EXISTS script_address ON script_transactions (scriptAddress)"
   -- Add this index for interval queries.
   SQL.execute_ c "CREATE INDEX IF NOT EXISTS script_address_slot ON script_transactions (scriptAddress, slotNo)"
-  emptyState k (ScriptTxHandle c (k * 2))
+  emptyState k (ScriptTxHandle c k)
```

Comment:
The second field of `ScriptTxHandle` is called `hdlDiskStore` but it's set to depth -- is the value of it always actually the depth? If so, maybe call it that as it's easier to understand.

Reply:
Good idea. I changed the name.